core/domain/exp_services.py

```diff
@@ -2990,22 +2963,23 @@ def is_version_of_draft_valid(exp_id: str, version: int) -> bool:
 
 
 def get_user_exploration_data(
-    user_id: str,
+    user_id: str | None,
```

Comment:
Don't loosen the type here; don't call this if user_id is None. Caller should handle that case. It makes no sense to get this object when there's no user ID to key it by.

Reply:
Done. Now we are not calling the function in case of a None user Id.